modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/QueueRegistry.java

```diff
@@ -14,111 +14,37 @@
  * KIND, either express or implied. See the License for the
  * specific language governing permissions and limitations
  * under the License.
- *
  */
 
 package io.ballerina.messaging.broker.core;
 
 import io.ballerina.messaging.broker.common.ResourceNotFoundException;
 import io.ballerina.messaging.broker.common.ValidationException;
-import io.ballerina.messaging.broker.core.store.dao.QueueDao;
+import io.ballerina.messaging.broker.common.data.types.FieldTable;
 
 import java.util.Collection;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Objects;
 
 /**
- * Registry object which contains all the queues of the broker.
+ * Abstract class for Queue Registry objects.
  */
-public final class QueueRegistry {
-
-    private final Map<String, QueueHandler> queueHandlerMap;
-
-    private final QueueDao queueDao;
-
-    private final QueueHandlerFactory queueHandlerFactory;
-
-    public QueueRegistry(QueueDao queueDao, QueueHandlerFactory queueHandlerFactory) throws BrokerException {
-        this.queueHandlerMap = new HashMap<>();
-        this.queueDao = queueDao;
-        this.queueHandlerFactory = queueHandlerFactory;
-        retrieveQueuesFromDao();
-    }
-
-    QueueHandler getQueueHandler(String queueName) {
-        return queueHandlerMap.get(queueName);
-    }
-
-    boolean addQueue(String queueName, boolean passive, boolean durable, boolean autoDelete) throws BrokerException {
-        QueueHandler queueHandler = queueHandlerMap.get(queueName);
-
-        if (passive) {
-            if (Objects.isNull(queueHandler)) {
-                throw new BrokerException("Queue [ " + queueName + " ] doesn't exists. Passive parameter "
-                                                  + "is set, hence not creating the queue.");
-            } else {
-                return false;
-            }
-        } else {
-            if (Objects.isNull(queueHandler)) {
-                if (durable) {
-                    queueHandler = queueHandlerFactory.createDurableQueueHandler(queueName, autoDelete);
-                    queueDao.persist(queueHandler.getUnmodifiableQueue());
-                } else {
-                    queueHandler = queueHandlerFactory.createNonDurableQueueHandler(queueName, autoDelete);
-                }
-                queueHandlerMap.put(queueName, queueHandler);
-                return true;
-            } else if (queueHandler.getUnmodifiableQueue().isDurable() != durable
-                       || queueHandler.getUnmodifiableQueue().isAutoDelete() != autoDelete) {
-                throw new BrokerException(
-                        "Existing queue [ " + queueName + " ] does not match given parameters.");
-            } else {
-                return false;
-            }
-        }
-    }
+public abstract class QueueRegistry {
```

Comment:
QueueRegistry is used in io.ballerina.messaging.broker.core.store.StoreFactory which is outside the package. That's why it is not possible to make it package private. And most of its methods are supposed to be used within the package, not in public. That's why I used an abstract class instead of an interface which has all methods public.